src/file_parsing.jl

```diff
@@ -151,35 +151,68 @@ function log_component!(
     end
 end
 
+"""
+    _get_typed_value(pd,val,valtype)
+
+enforces `val` to be of type as specified in the toml file
+- `float_type(pd)` if type=\"float\"
+- `Int` if type=\"integer\"
+- `String` if  type=\"string\"
+Default type of `String` is used if no type is provided.
+"""
+function _get_typed_value(pd, val, valtype)
```

Comment:
I'm wondering if we should also pass the key (i.e., `name` or `alias`) into this function so that we get a more informative error message if/when this errors. Users may not know which type is wrong only based on the value.

Reply:
Yeah I also thought this - would be happy to do so

Reply:
Although i would say the error message does tell you which "type" i.e the `type = "incorrect"` statement here. Not the value. (So it would be searchable)